### PR TITLE
feat: Vault server - adds authorizations

### PR DESCRIPTION
**Pull request by soluchok**

### CODE REVIEW

File: pkg/client/vault/client.go [hunk 1]
```diff
@@ -170,16 +203,84 @@ func (c *Client) CreateVault() (*CreatedVault, error) {
...
+
+	kmsCapability, err := uncompressZCAP(info.Auth.KMS.AuthToken)
+	if err != nil {
+		return nil, fmt.Errorf("kms uncompressZCAP: %w", err)
+	}
+
+	kmsNewCapability, err := zcapld.NewCapability(&zcapld.Signer{
+		SignatureSuite:     ed25519signature2018.New(suite.WithSigner(newSigner(c.crypto, kh))),
+		SuiteType:          ed25519signature2018.SignatureType,
+		VerificationMethod: vaultID,
```
Comment (llorllale) [inline pkg/client/vault/client.go:227]: `vaultID` must be a DID, not a DID URL

File: pkg/client/vault/client.go [hunk 2]
```diff
@@ -170,16 +203,84 @@ func (c *Client) CreateVault() (*CreatedVault, error) {
...
+	}, zcapld.WithParent(kmsCapability.ID), zcapld.WithInvoker(requestingParty),
+		zcapld.WithAllowedActions("unwrap"),
+		zcapld.WithInvocationTarget(kmsCapability.InvocationTarget.ID, kmsCapability.InvocationTarget.Type),
```
Comment (llorllale) [inline pkg/client/vault/client.go:230]: TODO: this is pointing to the keystore and not the specific key, should be the unwrapping key. Requires a change in hub-kms, likely a new endpoint to retrieve zcaps for more fine-grained resources.

File: pkg/client/vault/client.go [hunk 3]
```diff
@@ -170,16 +203,84 @@ func (c *Client) CreateVault() (*CreatedVault, error) {
...
+		return nil, fmt.Errorf("edv uncompressZCAP: %w", err)
+	}
+
+	edvNewCapability, err := zcapld.NewCapability(&zcapld.Signer{
+		SignatureSuite:     ed25519signature2018.New(suite.WithSigner(newSigner(c.crypto, kh))),
+		SuiteType:          ed25519signature2018.SignatureType,
+		VerificationMethod: vaultID,
+	}, zcapld.WithParent(edvCapability.ID), zcapld.WithInvoker(requestingParty),
+		zcapld.WithAllowedActions(scope.Actions...),
+		zcapld.WithInvocationTarget(edvCapability.InvocationTarget.ID, edvCapability.InvocationTarget.Type),
```
Comment (llorllale) [inline pkg/client/vault/client.go:252]: Same for the EDV (see https://github.com/trustbloc/edge-service/pull/601/files#r575187335)

**Review by llorllale — APPROVED** [review]
* need to fix the vaultID
* zcap authorizations need to be more fine-grained - we'll tune this after basic integration is accomplished

**Review by aholovko — APPROVED** [review]

**Pull request merged**